benchmark/CMakeLists.txt

```diff
@@ -29,7 +34,7 @@ foreach(benchmark_src ${rocRAND_BENCHMARK_SRCS})
     else()
         target_link_libraries(${BENCHMARK_TARGET}
             rocrand
-            hip::device
+            $<$<NOT:$<BOOL:ROCRAND_CMAKE_HIP_LANGUAGE>>:hip::device>
```

Comment:
Nit: for benchmarks an if statement is used, but here its a generator expression. I would stick to the if in both places for consistency, and because this setting is known at configure time, a generator expression is not required.